app/lib/dns.server.ts

```diff
@@ -157,8 +168,28 @@ export const getChangeStatus = async (changeId: string) => {
   }
 };
 
-const isNameValid = (name: string) => {
-  return /^[a-z0-9-]+.[a-z0-9-]+.[a-z]+.[a-z]+\.?$/.test(name);
+/* Domain name rules
+1. Domain name pattern should be [name].[username].rootDomain.com
+2. Domain name can contain only alphanumerical characters, '-', and '_'
+3. Domain name should not start or end with -
+4. Domain name cannot contain multiple consecutive '-' or '_'
+5. Domain name can contain uppercase in UI but it is converted to lowercase before validation */
+export const isNameValid = (name: string, username: string) => {
+  const rootDomain = process.env.ROOT_DOMAIN!;
+  let subdomain: string = '';
+
+  if (!name.endsWith(`.${username}.${rootDomain}`)) {
+    return false;
+  }
+  subdomain = name.replace(`.${username}.${rootDomain}`, '');
```

Comment:
```suggestion
  const toRemove = `.${username}.${rootDomain}`;
  if (!name.endsWith(toRemove)) {
    return false;
  }
  subdomain = name.substring(0, name.length - toRemove.length);
```
Sorry for picking on this, but I'd also make sure that this is removed at the end of the string